### ISSUE TITLE
Inline widget styling

### ISSUE THREAD

**Issue by dkrahn** [ckeditor/ckeditor5#1633]
## Is this a bug report or feature request? (choose one)

🐞 Bug report or documentation improvement

## 💻 Version of CKEditor

12.0.0

## 📋 Steps to reproduce

1. Follow the steps from the Inline Widget docs (https://ckeditor.com/docs/ckeditor5/latest/framework/guides/tutorials/implementing-an-inline-widget.html) or (https://ckeditor5.github.io/misc/placeholder-demo/index.html)
2. Try to style all the text

## ✅ Expected result
From my perspective, it would be desired of the inline widget to be treated as part of the text when it comes to styling(bold, italic, font...).

## ❎ Actual result
The text around the inline widget get styled but the widget itself does not.

## 📃 Other details that might be useful
I was able to get around this issue with changing the createContainerElement with createAttributeElement, adding a random id and priority 20.
And then adding the text using the private _insertChild converting elementToElement. But this does not seam right.

---

If you'd like to see this feature implemented, add 👍 to this post.

**Comment by Mgsy** [comment]
cc @jodator

**Comment by jodator** [comment]
Right now we do not support this out of the box. I think that view writers should be updated to support `isInline=true` elements in wrapping in attributes elements but I think that's a broader topic.

//cc @Reinmar & @scofalik.

**Comment by dkrahn** [comment]
Hi @jodator 
IMHO the outcome should be se second one:
```
<p>
	<strong>Bold 
	<placeholder>{place}</placeholder>
	.</strong>
</p>
```

And about using the private API, that is exactly why I opened this issue, it doesn't look right.

**Comment by Reinmar** [comment]
`<strong>` should definitely be outside `<placeholder>` in the view. It needs to be done in three steps:

1. First (simplest), you need to use `allowAttributesOf: '$text'` in your widget's schema definition.
2. Then, we need to make sure that basic styles correctly apply to such elements. Right now, they may be implemented in a way that they only look for text nodes. This needs to be generalised so they look for all inline nodes.
3. Finally, and this is the biggest challenge, if we'll want to render `<strong><placeholder/></strong>` in the view we need to render a `ContainerElement` inside an `AttributeElement` which isn't allowed today. Today, all the code assumes that there's only text or `UIElement` (and perhaps `EmptyElement`) in `AttributeElements`. We may need to either remove that limitation or introduce a new element type.

Anyway, tl;dr is that unfortunately this isn't supported yet and it may not be that easy to introduce (in a stable way).

**Comment by pragmaticus** [comment]
Any progress on this?

@jodator i couldn't reproduce the workaround with by using the private API. Can you post an example or guide me in the right direction?

**Comment by RemigijusValys** [comment]
Hello, any update on this ? 

Placeholders needs to be styled with font, bold, italic, etc. Their purpose is to provide a block that will be replaced with actual text. Without correct style they are not usable.

**Comment by Reinmar** [comment]
Not yet. It will be a pretty big change, unfortunately, so we need to find space for it in our roadmap. At the moment, I can't promise anything, but it's high on our wishlist too.

**Comment by nikunjtrapasiya** [comment]
Any progress on this?

**Comment by top-dev830** [comment]
Is there any update on this feature?

**Comment by Xusifob** [comment]
Hello,

Any information on this feature ? :)

**Comment by Xusifob** [comment]
I created a workaround for this. It's not optimal but instead of creating <strong> or <em> HTML elements inside the inline widget, I add a strong="true", italic="true" attributes in the widget element. Does the work for me.
You can check it out inside my fork

**Comment by dkrahn** [comment]
Any updates with the new version?

**Comment by QuantumGlitch** [comment]
Hi! I will be really interested in this feature too!

**Comment by dkrahn** [comment]
Hi,
Any estimates or updates on when this issue might get in the roadmap?
Thanks for the great work you all are doing!

**Comment by DaveOrDead** [comment]
Hey @dkrahn, hit the exact same situation following the tutorial. Are you able to elaborate on how you achieved your workaround as I haven't been able to figure out how to get it working. I wasn't exactly sure what this part meant `adding a random id and priority 20` 

thanks
